core/controllers/editor.py

```diff
@@ -933,3 +936,36 @@ class StartedTutorialEventHandler(EditorHandler):
     def post(self):
         """Handles GET requests."""
         user_services.record_user_started_state_editor_tutorial(self.user_id)
+
+
+class EditorAutosaveHandler(ExplorationHandler):
+    """Handles requests from the editor for draft autosave."""
+
+    @require_editor
+    def put(self, exploration_id):
+        """Handles PUT requests for draft updation."""
+        # Raise an Exception if the draft change list fails non-strict
+        # validation.
+        try:
+            change_list = self.payload.get('change_list')
+            version = self.payload.get('version')
+            exp_services.create_or_update_draft(
+                exploration_id, self.user_id, change_list, version,
+                datetime.datetime.utcnow())
+        except utils.ValidationError as e:
+            # If this exception is raised, let the user know that their changes
+            # have been auto discarded.
+            exp_services.discard_draft(exploration_id, self.user_id)
+            raise self.InvalidInputException(e)
+
+        # If the value passed here is False, have the user discard the draft
+        # changes.
+        self.render_json({
+            'is_draft_version_valid': exp_services.is_draft_version_valid(
```

Comment:
Ignore the previous comment. I had a discussion with Sean about this and I switched back to passing exploration_id instead of the exploration. 
@526avijitgupta FYI for you.
One thing we realized during our discussions is that the method '_get_exploration_data()' is also used by the history tab. When the history tab uses it though, it also passes an optional version param. We have to be aware of two things here:
1) We don't check version validity when the history tab is loaded.
2) We don't apply drafts when the history tab is loaded.
To do that we need to differentiate between when the method call comes from the history tab and when it comes from the editor tab. I'm gonna add an optional param  ExplorationHandler.get to do that.